doc/Makefile

```diff
@@ -134,10 +134,13 @@ pdf: latex
 
 install:
 	rm -rf _build/doctrees/
+	# first clone the nilearn.github.io repo because it may ask
+	# for password and we don't want to delay this long build in
+	# the middle of it
```

Comment:
Please comment on what the no-checkout does and why it's useful.

Also, please rebase to master to have circle-ci, so that we can test this with the doc building.